### PR TITLE
Aoc/623 slice long name elipsis

### PR DESCRIPTION
**Pull request by chengoconnell**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

https://user-images.githubusercontent.com/69536010/176164648-cab7fe3f-b4d8-427f-b3ee-05ebbede377e.mov

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/lib/models/ui/Slice/index.tsx [hunk 1]
```diff
@@ -178,9 +183,9 @@ export const SharedSlice = {
                   ) : null}
                 </Fragment>
               )}
-              <Heading sx={{ flex: 1 }} as="h6">
-                {slice.model.name}
-              </Heading>
+              <Flex sx={{ minWidth: 0 }}>
```
Comment (sintinilorenzo) [inline packages/slice-machine/lib/models/ui/Slice/index.tsx]: is this minWidth:0 needed?

**Review by sintinilorenzo — APPROVED** [review]

File: packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <>
+      <Heading
+        data-for={text}
+        data-tip
+        sx={{
+          flex: 1,
+          overflow: "hidden",
+          textOverflow: "ellipsis",
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx:28]: nice...

File: packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      >
+        {text}
+      </Heading>
+      {showTooltip ? (
+        <ReactTooltip
+          id={text}
+          type="dark"
+          multiline
+          border
+          borderColor={"black"}
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx]: why are props sometimes strings wrapped in `{..}` and other times just strings?

File: packages/slice-machine/lib/models/ui/Slice/index.tsx [hunk 4]
```diff
@@ -64,7 +65,7 @@ const SliceVariations = ({
   return !hideVariations ? (
     <>
       {variations ? (
-        <Text sx={{ fontSize: 0, color: "textClear" }}>
+        <Text sx={{ fontSize: 0, color: "textClear", flexShrink: 0 }}>
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/models/ui/Slice/index.tsx:68]: I guess this is in a flexbox somewhere

File: packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx (hunk 2 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx:28]: how does shotTooltip go back to false?
does it need to?

**Review by MarcMcIntosh — COMMENTED** [review]
A few questions :)

File: packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx (hunk 3 above)
Comment (chengoconnell) [reply packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx]: this is a good point. Lemme update to maintain consistency!

File: packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx (hunk 2 above)
Comment (chengoconnell) [reply packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx:28]: if `headingRef.current.offsetWidth < headingRef.current.scrollWidth` = false, it'll set it back to false :) 
i believe:
offsetWidth = width on display
scrollWidth = width on display + hidden

Comment (MarcMcIntosh) [reply packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx:28]: does it work when you hover over the same ones a few times?

Comment (chengoconnell) [reply packages/slice-machine/components/Tooltip/HeadingWithTooltip.tsx:28]: yup :)

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**